test/test-index.ts

```diff
@@ -102,21 +90,40 @@ test('should return only the first error even if multiple would throw', (t) => {
       el.setAttribute('rendered', '');
     },
     'amp-fail1': (_el: Element) => {
-      throw new Error('Cannot render <amp-fail1>');
+      throw new Error('Something is undefined');
     },
     'amp-fail2': (_el: Element) => {
-      throw new Error('Cannot render <amp-fail2>');
+      throw new Error('Something is undefined');
     },
   };
 
   const ast = treeProto(
     h('amp-success', {}, [h('amp-fail1'), h('amp-fail1'), h('amp-fail2')])
   );
-  const expected: Result<string> = {
-    type: 'failure',
-    error: new Map([['amp-fail1', ['Cannot render <amp-fail1>']]]),
+  t.throws(() => renderAst(ast, instructions), {message: /amp-fail/});
```

Comment:
This matcher would pass even if the second error was being thrown. Consider something like `amp-first` and `amp-second` to distinguish

Reply:
Great point. Updating